common/src/test/java/com/revenuecat/purchases/common/HTTPClientVerificationTest.kt

```diff
@@ -175,14 +175,22 @@ class HTTPClientVerificationTest: BaseHTTPClientTest() {
             mockSigningManager.verifyResponse(any(), any(), any(), any(), any(), any(), any())
         } returns VerificationResult.FAILED
 
-        client.performRequest(
-            baseURL,
-            endpoint,
-            body = null,
-            requestHeaders = emptyMap()
-        )
+        var thrownCorrectException = false
+        try {
+            client.performRequest(
+                baseURL,
+                endpoint,
+                body = null,
+                requestHeaders = emptyMap()
+            )
+        } catch (_: SignatureVerificationException) {
+            thrownCorrectException = true
+        }
 
-        server.takeRequest()
+        assertThat(thrownCorrectException).isTrue
+        verify(exactly = 0) {
+            mockETagManager.getHTTPResultFromCacheOrBackend(any(), any(), any(), any(), any(), any())
+        }
```

Comment:
Improved this test to make sure we don't check with etag manager when verification errors and verification mode is `enforced`.